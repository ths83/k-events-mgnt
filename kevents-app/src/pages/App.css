.layout {
  min-height: 100vh;
}

.logo {
  height: 2rem;
  margin: 1rem;
}

.appName {
  color: #fff;
  display: flex;
  justify-content: center;
}

.site-layout .site-layout-background {
  background: #fff;
}

.content {
  margin: 1rem;
}

.footer {
  text-align: center;
}
