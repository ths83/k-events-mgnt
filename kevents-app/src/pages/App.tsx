import { AlignLeftOutlined, PlusOutlined } from "@ant-design/icons";
import { Menu } from "antd";
import Layout, { Content, Footer } from "antd/lib/layout/layout";
import Sider from "antd/lib/layout/Sider";
import Text from "antd/lib/typography/Text";
import React, { useState } from "react";
import { MENU_OPTIONS } from "../constants/MenuConstants";
import "./App.css";

const App = () => {
  const [collapsed, setCollapse] = useState(false);
  const [defaultOption, setDefaultOption] = useState(MENU_OPTIONS.CREATE);

  return (
    <Layout className="layout">
      <Sider
        collapsible
        collapsed={collapsed}
        onCollapse={() => setCollapse(!collapsed)}
      >
        <div className="logo">
          <Text id="appName" className="appName">
            KEvents
          </Text>
        </div>
        <Menu theme="dark" defaultSelectedKeys={[defaultOption]} mode="inline">
          <Menu.Item
            id="newEventOption"
            key={MENU_OPTIONS.CREATE}
            icon={<PlusOutlined />}
            onClick={() => setDefaultOption(MENU_OPTIONS.CREATE)}
          >
            New event
          </Menu.Item>
          <Menu.Item
            id="listEventsOption"
            key={MENU_OPTIONS.LIST}
            icon={<AlignLeftOutlined />}
            onClick={() => setDefaultOption(MENU_OPTIONS.LIST)}
          >
            List events
          </Menu.Item>
        </Menu>
      </Sider>
      <Layout className="site-layout">
        <Content className="content">
          {defaultOption === MENU_OPTIONS.CREATE ? (
            <p>This is text 1</p>
          ) : (
            <p>This is text 2</p>
          )}
        </Content>
        <Footer id="footer" className="footer">
          Â©2022 Created by Thomas Perron
        </Footer>
      </Layout>
    </Layout>
  );
};

export default App;
