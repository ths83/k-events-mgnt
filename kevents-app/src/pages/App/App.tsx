import { AlignLeftOutlined, PlusOutlined } from "@ant-design/icons";
import { Menu } from "antd";
import Layout, { Content, Footer } from "antd/lib/layout/layout";
import Sider from "antd/lib/layout/Sider";
import Text from "antd/lib/typography/Text";
import React, { useEffect, useState } from "react";
import Events from "../../components/events/Events";
import NewEventForm from "../../components/newEventForm/NewEventForm";
import { clearEvents } from "../../features/eventsSlice";
import "./App.css";

const App = () => {
  const menuOptions = {
    CREATE: "1",
    LIST: "2",
  };

  const [collapsed, setCollapse] = useState(false);
  const [defaultOption, setDefaultOption] = useState(menuOptions.CREATE);

  useEffect(() => {
    clearEvents();
  }, [defaultOption]);

  return (
    <Layout className="layout">
      <Sider
        collapsible
        collapsed={collapsed}
        onCollapse={() => setCollapse(!collapsed)}
      >
        <div className="logo">
          <Text id="appName" className="appName">
            KEvents
          </Text>
        </div>
        <Menu theme="dark" defaultSelectedKeys={[defaultOption]} mode="inline">
          <Menu.Item
            id="newEventOption"
            key={menuOptions.CREATE}
            icon={<PlusOutlined />}
            onClick={() => setDefaultOption(menuOptions.CREATE)}
          >
            New event
          </Menu.Item>
          <Menu.Item
            id="listEventsOption"
            key={menuOptions.LIST}
            icon={<AlignLeftOutlined />}
            onClick={() => setDefaultOption(menuOptions.LIST)}
          >
            List events
          </Menu.Item>
        </Menu>
      </Sider>
      <Layout className="site-layout">
        <Content className="content">
          {defaultOption === menuOptions.CREATE ? <NewEventForm /> : <Events />}
        </Content>
        <Footer id="footer" className="footer">
          Â©2022 Created by Thomas Perron
        </Footer>
      </Layout>
    </Layout>
  );
};

export default App;
