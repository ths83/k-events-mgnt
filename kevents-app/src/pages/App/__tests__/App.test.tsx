import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../App";
import renderer from "react-test-renderer";

const expectedAppName = "KEvents";
const expectedNewEventOption = "New event";
const expectedListEventsOption = "List events";
const expectedFooterContent = "Â©2022 Created by Thomas Perron";

test("renders component", () => {
  const item = renderer.create(<App />).toJSON();
  expect(item).toMatchSnapshot();
});

test("renders menu", () => {
  render(<App />);
  const appName = screen.getByText(expectedAppName);
  const newEventMenuOption = screen.getByText(expectedNewEventOption);
  const listEventsMenuOption = screen.getByText(expectedListEventsOption);

  expect(appName).toBeInTheDocument();
  expect(newEventMenuOption).toBeInTheDocument();
  expect(listEventsMenuOption).toBeInTheDocument();
});

test("renders footer", () => {
  render(<App />);
  const footer = screen.getByText(expectedFooterContent);

  expect(footer).toBeInTheDocument();
});
