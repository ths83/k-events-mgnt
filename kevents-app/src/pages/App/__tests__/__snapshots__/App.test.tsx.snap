// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<section
  className="ant-layout ant-layout-has-sider layout"
>
  <aside
    className="ant-layout-sider ant-layout-sider-dark ant-layout-sider-has-trigger"
    style={
      Object {
        "flex": "0 0 200px",
        "maxWidth": "200px",
        "minWidth": "200px",
        "width": "200px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <div
        className="logo"
      >
        <span
          className="ant-typography appName"
          id="appName"
          onClick={[Function]}
          style={
            Object {
              "WebkitLineClamp": undefined,
            }
          }
        >
          KEvents
        </span>
      </div>
      <ul
        className="ant-menu ant-menu-root ant-menu-inline ant-menu-dark"
        data-menu-list={true}
        onKeyDown={[Function]}
        role="menu"
        tabIndex={0}
      >
        <li
          className="ant-menu-item ant-menu-item-selected"
          data-menu-id="rc-menu-uuid-test-1"
          id="newEventOption"
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
          tabIndex={-1}
        >
          <span
            aria-label="plus"
            className="anticon anticon-plus ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="plus"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <defs>
                <style />
              </defs>
              <path
                d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
              />
              <path
                d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
              />
            </svg>
          </span>
          <span
            className="ant-menu-title-content"
          >
            New event
          </span>
        </li>
        <li
          className="ant-menu-item"
          data-menu-id="rc-menu-uuid-test-2"
          id="listEventsOption"
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
          tabIndex={-1}
        >
          <span
            aria-label="align-left"
            className="anticon anticon-align-left ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="align-left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M120 230h496c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm0 424h496c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm784 140H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0-424H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
          <span
            className="ant-menu-title-content"
          >
            List events
          </span>
        </li>
      </ul>
      <div
        aria-hidden={true}
        style={
          Object {
            "display": "none",
          }
        }
      />
    </div>
    <div
      className="ant-layout-sider-trigger"
      onClick={[Function]}
      style={
        Object {
          "width": "200px",
        }
      }
    >
      <span
        aria-label="left"
        className="anticon anticon-left"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="left"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
          />
        </svg>
      </span>
    </div>
  </aside>
  <section
    className="ant-layout site-layout"
  >
    <main
      className="ant-layout-content content"
    >
      <h3
        className="ant-typography"
        onClick={[Function]}
        style={
          Object {
            "WebkitLineClamp": undefined,
          }
        }
      >
        New event form
      </h3>
      <form
        action="#"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="input"
        >
          <h5
            className="ant-typography"
            id="eventNameTitle"
            onClick={[Function]}
            style={
              Object {
                "WebkitLineClamp": undefined,
              }
            }
          >
            Name
          </h5>
          <input
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter an event name"
            value=""
          />
        </div>
        <div
          className="input"
        >
          <h5
            className="ant-typography"
            id="eventDescTitle"
            onClick={[Function]}
            style={
              Object {
                "WebkitLineClamp": undefined,
              }
            }
          >
            Description
          </h5>
          <input
            name="description"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Enter an event description"
            value=""
          />
        </div>
        <div
          className="input"
        >
          <h5
            className="ant-typography"
            id="eventStartDateTitle"
            onClick={[Function]}
            style={
              Object {
                "WebkitLineClamp": undefined,
              }
            }
          >
            Start Date
          </h5>
          <div
            className="ant-picker"
            onMouseUp={[Function]}
          >
            <div
              className="ant-picker-input"
            >
              <input
                autoComplete="off"
                id="startDate"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                placeholder="Select date"
                readOnly={true}
                size={18}
                title=""
                value=""
              />
              <span
                className="ant-picker-suffix"
              >
                <span
                  aria-label="calendar"
                  className="anticon anticon-calendar"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="calendar"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          className="input"
        >
          <h5
            className="ant-typography"
            id="eventEndDateTitle"
            onClick={[Function]}
            style={
              Object {
                "WebkitLineClamp": undefined,
              }
            }
          >
            End Date
          </h5>
          <div
            className="ant-picker"
            onMouseUp={[Function]}
          >
            <div
              className="ant-picker-input"
            >
              <input
                autoComplete="off"
                id="endDate"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                placeholder="Select date"
                readOnly={true}
                size={18}
                title=""
                value=""
              />
              <span
                className="ant-picker-suffix"
              >
                <span
                  aria-label="calendar"
                  className="anticon anticon-calendar"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="calendar"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          className="buttons"
        >
          <button
            className="ant-btn ant-btn-primary ant-btn-round ant-btn-dangerous"
            id="reset"
            onClick={[Function]}
            type="button"
          >
            <span>
              Reset
            </span>
          </button>
          <button
            className="ant-btn ant-btn-primary ant-btn-round"
            disabled={true}
            id="submit"
            onClick={[Function]}
            type="button"
          >
            <span>
              Submit
            </span>
          </button>
        </div>
      </form>
    </main>
    <footer
      className="ant-layout-footer footer"
      id="footer"
    >
      ©2022 Created by Thomas Perron
    </footer>
  </section>
</section>
`;
